app/javascript/react/components/DeleteButton.js

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+
+const DeleteButton = props => {
+  const {role, id} = props
+
+  const render = () => {
+    props.render()
+  }
+
+  const deleteMonsterFetch = async () => {
+    try {
+      const response = await fetch(`/api/v1/monsters/${id}`, {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "application/json",
+          "Accept": "application/json"
+        },
+        credentials: 'same-origin'
+      })
+      if(!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`)
+      }
+      render()
```

Comment:
While re-fetching provides us with updated data, we can avoid an extra fetch request by using the `id` above to remove that monster from state of all monsters